go-controller/pkg/types/const.go

```diff
@@ -53,4 +53,10 @@ const (
 
 	V4JoinSubnetCIDR = "100.64.0.0/16"
 	V6JoinSubnetCIDR = "fd98::/64"
+
+	// OVNKube-Node Node types
+	NodeModeFull = "full"
+	NodeModeSmartNIC = "smart-nic"
+	NodeModeSmartNICHost = "smart-nic-host"
```

Comment:
totally naive in this field, what are the main differences of each mode?

Reply:
Hi, I added some info in one of the commit messages:

```
According to the mode ovnkube-node-mode is set, a different
initialization flow will occur on Start() method.

mode="full"
  No modification to ovnkube Node Start() logic

mode=smart-nic-host:
  This initial separation will essentially perform the same
  logic as was perfomed on the host by the removed smart-nic-cni
  via cniserver in `smart-nic-host` mode. ovnkube Node Start()
  method in this mode will essentialy just setup  logging and
  create and run cniserver

mode=smart-nic:
  Perform the same initialization during ovnkube Node Start()
  however, do not start cni server and instead watch for
  pod events to peform the neccessary plumbing on the smart-nic
  according to special smart-nic anntoation on pod.

Note:
  Subsequent commits will further separate ovnkube Node logic
  between smart-nic and smart-nic-host. e.g. gateway initialization
  management port initialization.
```

Reply:
Can we add this as comments in the code, it will be easier to find and to understand for new people to this topic or to the project, ... also for future us :smile: